### PR TITLE
(freecad) Migrated to AU

### PR DESCRIPTION
**Pull request by AdmiringWorm**
### Other Changes
* added project source url
* updated icon url
* added generation of release notes url
* updated package source url
* updated install script
* Added uninstall script (auto uninstaller didn't uninstall application)

**NOTE** Needs force push

also fixes issue #251

### CODE REVIEW

File: automatic/freecad/update.ps1 [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    ".\tools\chocolateyInstall.ps1" = @{
+      "(?i)(^\s*url\s*=\s*)('.*')"        = "`$1'$($Latest.URL32)'"
+      "(?i)(^\s*url64\s*=\s*)('.*')"        = "`$1'$($Latest.URL64)'"
+      "(?i)(^\s*checksum\s*=\s*)('.*')"   = "`$1'$($Latest.Checksum32)'"
+      "(?i)(^\s*checksum64\s*=\s*)('.*')"   = "`$1'$($Latest.Checksum64)'"
+      "(?i)(^\s*checksumType\s*=\s*)('.*')" = "`$1'$($Latest.ChecksumType32)'"
+      "(?i)(^\s*checksumType64\s*=\s*)('.*')" = "`$1'$($Latest.ChecksumType64)'"
+    }
+    ".\freecad.nuspec" = @{
+      "\<releaseNotes\>.+" = "<releaseNotes>$($Latest.ReleaseNotes)</releaseNotes>"
```
Comment (majkinetor) [inline automatic/freecad/update.ps1]: While this works in this specific case its a wrong habit to have - nuspec can have multiple tags on the single line. Some other user may have problems or you in the future.

Here is my template:

```

function global:au_SearchReplace {
   @{
        ".\tools\chocolateyInstall.ps1" = @{
            "(?i)(^\s*url\s*=\s*)('.*')"          = "`$1'$($Latest.URL32)'"
            "(?i)(^\s*url64bit\s*=\s*)('.*')"     = "`$1'$($Latest.URL64)'"
            "(?i)(^\s*checksum\s*=\s*)('.*')"     = "`$1'$($Latest.Checksum32)'"
            "(?i)(^\s*checksum64\s*=\s*)('.*')"   = "`$1'$($Latest.Checksum64)'"
            "(?i)(^\s*packageName\s*=\s*)('.*')"  = "`$1'$($Latest.PackageName)'"
            "(?i)(^\s*softwareName\s*=\s*)('.*')" = "`$1'$($Latest.PackageName)*'"
            "(?i)(^\s*fileType\s*=\s*)('.*')"     = "`$1'$($Latest.FileType)'"
        }

        "$($Latest.PackageName).nuspec" = @{
            "(\<releaseNotes\>).*?(\</releaseNotes\>)" = "`${1}$($Latest.ReleaseNotes)`$2"
        }
    }
}
```

Here you can see that it starts from releaseNotes tag, it takes non greedy everything up to the closing tag.

Comment (AdmiringWorm) [inline automatic/freecad/update.ps1]: ah, didn't think of that. I'm on my way out, but I'll change it when I come back

Comment (majkinetor) [inline automatic/freecad/update.ps1]: You don't have to change it for this package, just an info for the future.

**Comment by majkinetor** [comment]
It doesn't uninstall on its own ?

**Comment by AdmiringWorm** [comment]
@majkinetor when I tested it, the auto uninstaller didn't detect the uninstaller.
Just reported it already was uninstalled.

**Pull request merged**